homeassistant/components/aemet/weather_update_coordinator.py

```diff
@@ -95,15 +97,15 @@ def format_condition(condition: str) -> str:
     return condition
 
 
-def format_float(value) -> float:
+def format_float(value) -> float | None:
     """Try converting string to float."""
     try:
         return float(value)
     except (TypeError, ValueError):
         return None
 
 
-def format_int(value) -> int:
+def format_int(value) -> int | None:
```

Comment:
Please change the methods where format_int is used, to ensure the check is checking something.

Reply:
I don't see why that would be necessary? If the value is None `format_int` should return None.

Reply:
That is not “nessecary” to stop mypy from complaining, but that is not a goal! the goal is to correct the code and make it strongly typed. When you declare that a function can return None, it is nice to know you have considered the places where it is used.

Reply:
> That is not “nessecary” to stop mypy from complaining, but that is not a goal! the goal is to correct the code and make it strongly typed.

I totally agree.

> When you declare that a function can return None, it is nice to know you have considered the places where it is used.

I did. If it can't cast the value to an int **or** if the value is already None the function should return None. If it can cast the value it should return the value as int.

Reply:
You did not change the functions where it is used.